File: problems/2309-greatest-english-letter-in-upper-and-lower-case/solution.py
class Solution:
    def greatestLetter(self, s: str) -> str:
        letters = set(s)
        for char_code in range(ord('Z'), ord('A') - 1, -1):
            char = chr(char_code)
            if char in letters and char.lower() in letters:
                return char
        return ""

File: problems/2309-greatest-english-letter-in-upper-and-lower-case/solution.cpp
class Solution {
public:
    string greatestLetter(string s) {
        bool lower[26] = {false};
        bool upper[26] = {false};

        for (char c : s) {
            if (islower(c)) {
                lower[c - 'a'] = true;
            } else {
                upper[c - 'A'] = true;
            }
        }

        for (int i = 25; i >= 0; --i) {
            if (lower[i] && upper[i]) {
                return string(1, (char)('A' + i));
            }
        }

        return "";
    }
};